src/js/components/chat-input.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { sendMessage } from 'services/conversation-service';
+
+export class ChatInputComponent extends Component {
+    constructor(...args) {
+        super(...args);
+
+        this.state = {
+            text: ''
+        };
+
+        this.onChange = this.onChange.bind(this);
+        this.sendMessage = this.sendMessage.bind(this);
+    }
+
+    onChange(e) {
+        this.setState({
+            text: e.target.value
+        });
+    }
+
+    sendMessage(e) {
+        e.preventDefault();
+        const text = this.state.text;
+        if (!!text.trim()) {
+            this.setState({
+                text: ''
+            });
+            sendMessage(text);
```

Comment:
This is pretty confusing, looks like a recursive call. Any way to make it clearer?